platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ProblemLookup.java

```diff
@@ -16,22 +16,8 @@
 
 package org.gradle.api.problems.internal;
 
-import org.gradle.api.NonNullApi;
-
 import java.util.Collection;
 
-/**
- * If implemented on an {@link Exception}, this interface will be used to fetch and render
- * problems specific to the exception.
- */
-@NonNullApi
-public interface ProblemAwareFailure {
-
-    /**
-     * Returns a collection of problems that are specific to this exception.
-     *
-     * @return a collection of problems
-     * @since 8.10
-     */
-    Collection<Problem> getProblems();
+public interface ProblemLookup {
```

Comment:
🤔 Something doesn't click with me with this classname, but I cannot come up with something better either.
Any case, this is internal so if we come up with something better, we can change it anytime.

Reply:
ProblemLocator?